### DISCUSSION TITLE
Support MacOS versions before 13.0

### DISCUSSION THREAD

**Discussion by mlindner** [ghostty-org/ghostty#3696] [Feature Requests, Ideas] [answered]
13.0 is a really recent OS version. I'm on 12.7.6 and it says it's unsupported. A simple tty shouldn't be using such a recent OS version requirement. Support should be given for at least the MacOS versions that have official Apple support and OS versions that recently had support.

**Comment by JakeWharton** [comment]
> Support should be given for at least the MacOS versions that have official Apple support and OS versions that recently had support.

That is already the case today. macOS 12 is end of life as of November 2024.

https://endoflife.date/macos

**Reply by mlindner** [reply]
Last month is "recently had support"

**Reply by JakeWharton** [reply]
What is the cutoff for recent, then? The end of life cutoff is easy to use. Recent is arbitrary.

**Reply by mlindner** [reply]
I'm just saying at least the 1.0.0 release should have supported it, given it would've been under development in November when it was still supported.

**Reply by JakeWharton** [reply]
True. Although the December public release was known back then, too.

**Comment by mitchellh** [comment, marked as answer]
No, this is additional maintenance burden that I'm not interested in. If you'd like to take on the burden, please fork the project and maintain it yourself. As another option, you can find an old enough version of Ghostty in the history that supports macOS 12 if you'd like.

I'm not going to commit to always dropping support when Apple does EOL, but macOS 12 was a particularly special case because it was _riddled_ with bugs. We had many workarounds for macOS 12 and there were still a dozen or so other UI bugs within AppKit. There was also a Metal GPU driver bug on Intel machines that caused some issues. 

Of all the OS versions we've developed against so far (12, 13, 14, 15), macOS 12 has by far been the worst. I recommend upgrading if you can (for this but also just so you can get security updates, I'm not sure why you'd run an OS that has active published CVEs). 

I'm not going to support macOS versions before 13. I may continue supporting 13 when it's EOL; it's been quite stable. We'll see.

**Reply by mlindner** [reply]
That's unfortunate, but MacOS 12 has been great for me and I intentionally chose not to update because 13 brought many changes that are negative including a large focus on being a "mobile first" operating system. I guess I'll just continue to use other terminal emulators.

**Reply by mitchellh** [reply]
That's the correct choice if macOS 13+ is not an option for you. I'm sorry, but macOS 12 was hell for me as an app developer and I don't want to bring that back. It isn't much work in a fork to get Ghostty building for macOS 12 again, but there are various subtle UI bugs that do exist on that macOS version.

**Reply by mlindner** [reply]
Do you know which version was the last to support MacOS 12?

**Reply by jcollie** [reply]
PR #2429 is where macOS 12 support was removed.

**Reply by fightingdreamer** [reply]
@mlindner you can build Ghostty from my [fork](https://github.com/fightingdreamer/ghostty/tree/macos-12).

**Reply by mlindner** [reply]
@fightingdreamer Awesome! Thank you!

**Reply by evrial** [reply]
You're saying that like that's a Windows OS or Linux that can run on anything. MacOS 11 and 12 is dead stop for some machines with 16gb of RAM which isn't ancient by any means lol. And you leave them behind for e-waste, good for you.

**Reply by mitchellh** [reply]
> You're saying that like that's a Windows OS or Linux that can run on anything. MacOS 11 and 12 is dead stop for some machines with 16gb of RAM which isn't ancient by any means lol. And you leave them behind for e-waste, good for you.

Nothing is stopping you from forking it and supporting it like @fightingdreamer.

And I'm not creating any e-waste, there is plenty of terminal choice for people on those macOS versions.

I have no obligation.

**Comment by jcollie** [comment]
Locking this. The position of the Ghostty project w. r. t. macOS 12 has been made clear.
